### DISCUSSION TITLE
Question about Variance Settings in e3nn Library

### DISCUSSION THREAD

**Discussion by clecust** [e3nn/e3nn#393] [Q&A]
Hello,

I have been exploring the e3nn library and noticed that there are several variance settings, such as in1_var, in2_var, and out_var in o3.TensorProduct input. I am curious about the reason for these variance settings. Is it to normalize the data so that the neural network is better trained, or is it to preserve the equivariant properties of the network?

I would appreciate any insights or explanations regarding the purpose of these variance settings.

Thank you for your time and assistance.

Best regards, 
Junbao

**Comment by mariogeiger** [comment]
It's for normalization purposes. 

Here is the "user contract" with tensor product:

If the distributions of the inputs are independent and
If input 1 (2) has the distribution of a gaussian with mean=0 and variance = in1_var (in2_var), 
Then the output distribution has mean=0 and variance = out_var
